crates/sui-types/src/messages.rs

```diff
@@ -664,13 +674,13 @@ impl Transaction {
 
 impl Hash for Transaction {
     fn hash<H: Hasher>(&self, state: &mut H) {
-        self.data.hash(state);
+        self.signed_data.data.hash(state);
     }
 }
 
 impl PartialEq for Transaction {
     fn eq(&self, other: &Self) -> bool {
-        self.data == other.data
+        self.signed_data.data == other.signed_data.data
```

Comment:
Why define equality only on data if we have decided that now the digest will be over both data and user signature?

Reply:
@gdanezis is right, with non-malleable sigs, a tx is now defined by the whole `signed_data`. 

Note that an EdDSA or ECDSA user could submit the same tx `data` with two different valid sigs, that will result in our point of view to two different txIDs, and thus we consider them different txs.

Reply:
I see, as they produce different digests, they definitely probably also be considered different.